src/main/kotlin/org/rust/lang/core/macros/proc/ProcMacroServerPool.kt

```diff
@@ -65,18 +66,35 @@ class ProcMacroServerPool private constructor(
     override fun dispose() {}
 
     companion object {
-        fun tryCreate(toolchain: RsToolchainBase, parentDisposable: Disposable): ProcMacroServerPool? {
-            val expanderExecutable = RsPathManager.nativeHelper(toolchain is RsWslToolchain)
-            if (expanderExecutable == null || !expanderExecutable.isExecutable()) {
-                return null
+        fun new(toolchain: RsToolchainBase, expanderExecutable: Path, parentDisposable: Disposable): ProcMacroServerPool {
+            return ProcMacroServerPool(toolchain, expanderExecutable)
+                .also { Disposer.register(parentDisposable, it) }
+        }
+
+        fun findExpanderExecutablePath(toolchain: RsToolchainBase, sysroot: String): Path? {
+            return findExpanderFromToolchain(toolchain, sysroot) ?: findEmbeddedExpander(toolchain)
+        }
+
+        private fun findExpanderFromToolchain(toolchain: RsToolchainBase, sysroot: String): Path? {
+            val isWslToolchain = toolchain is RsWslToolchain
+            val binaryName = if (SystemInfo.isWindows && !isWslToolchain) {
+                "rust-analyzer-proc-macro-srv.exe"
+            } else {
+                "rust-analyzer-proc-macro-srv"
+            }
+
+            val expanderPath = Path.of(sysroot, "libexec", binaryName)
+            if (isWslToolchain) {
+                // Path.isExecutable() doesn't work for WSL files
+                if (!expanderPath.toFile().isFile) return null
+            } else {
+                if (!expanderPath.isExecutable()) return null
             }
-            return createUnchecked(toolchain, expanderExecutable, parentDisposable)
+            return expanderPath
         }
 
-        @VisibleForTesting
-        fun createUnchecked(toolchain: RsToolchainBase, expanderExecutable: Path, parentDisposable: Disposable): ProcMacroServerPool {
-            return ProcMacroServerPool(toolchain, expanderExecutable)
-                .also { Disposer.register(parentDisposable, it) }
+        private fun findEmbeddedExpander(toolchain: RsToolchainBase): Path? {
+            return RsPathManager.nativeHelper(toolchain is RsWslToolchain)
```

Comment:
Should we check executable here as before?
```suggestion
            return RsPathManager.nativeHelper(toolchain is RsWslToolchain)
                ?.takeIf { it.isExecutable() }
```

Reply:
It's now checked inside `RsPathManager.nativeHelper()`